lib/chef/resource.rb

```diff
@@ -1071,14 +1091,34 @@ def self.default_action=(action_name)
     #
     # @return The Action class implementing the action
     #
-    def self.action(action, &recipe_block)
+    def self.action(action, description: nil, &recipe_block)
       action = action.to_sym
       declare_action_class
       action_class.action(action, &recipe_block)
       self.allowed_actions += [ action ]
+      # TODO So this is weird and makes me think we shouldn't accept description in
+      # allowed_actions:  if someone for some reason has an action with a description,
+      # and allowed_actions also gives a description for the same action - which
+      # description prevails is load order dependent.
+      #
+      # In any case, accept any non-nil description, which will correctly override
+      # any specific inherited description.
```

Comment:
I definitely thing that the description should be with the action and we should not accept descriptions in `allowed_actions`. Allowed_actions - by it's very name, is a list of existing actions, and nothing more. Actions themselves should define their description just like resources themselves can.

Reply:
That makes sense. The reason I included it there in the first place is that  `allowed_actions` itself will otherwise add the named actions as valid actions - allowing an action to be declared without ever defining it.

Reply:
As long as we maintain our ability to document both custom resources and LWRP/HWRP style resources it's fine with me to drop this.